cmd/podman/images/scp.go

```diff
@@ -0,0 +1,383 @@
+	}
+	if scpOpts.ToFrom {
+		abiErr := abiEng.Save(context.Background(), args[0], tags, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+	}
+	uri, err := urlP.Parse(url) // create an actual url to pass to exec command
+	if err != nil {
+		return err
+	}
+	if uri.User.Username() == "" {
+		if uri.User, err = getUserInfo(uri); err != nil {
+			return err
+		}
+	}
+	rem := cmd.Root().Flags().Lookup("remote").Value // if remote was specified globally...
+	remS := rem.String()
+	if remS == "true" {
+		tunEng, err := infra.NewImageEngine(&optionsTunnel) // means we are doing a tunnel load
+		if err != nil {
+			return (errors.Wrapf(err, "could not make tunnel engine"))
+		}
+		tunOut, err := tunEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			errors.Wrapf(err, "could not load saved image")
+		}
+		if tunOut != nil {
+			out := tunOut.Names
+			if len(tunOut.Names) == 0 {
+				return errors.Wrapf(errors.New("Invalid Image given"), "could not get image")
+			}
+			fmt.Println("Loaded image(s): " + strings.Join(out, ","))
+		}
+	} else if len(args) == 2 && scpOpts.ToFrom { // else if a remote ssh connection was specified, means we need to go to exec command
+		rep, err := execCommand(iden, uri, scpOpts.Save.Output, scpOpts.Load.Input, args[0], scpOpts.ToFrom)
+		if err != nil {
+			return err
+		}
+		fmt.Println(rep)
+	} else if !scpOpts.ToFrom { // else if we want to load FROM the remote
+		_, err := execCommand(iden, uri, scpOpts.Save.Output, scpOpts.ImageName, args[0], scpOpts.ToFrom)
+		if err != nil {
+			return err
+		}
+		if len(args) == 2 && !(strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) { // if we don't want to do a remote -> remote then simply load
+			report, err := abiEng.Load(context.Background(), scpOpts.Load)
+			if err != nil {
+				return err
+			}
+			fmt.Println("Loaded images(s): " + strings.Join(report.Names, ","))
+		} else { // else we want to load remote -> remote
+			rep, err := execCommand(iden, uri, scpOpts.Save.Output, scpOpts.Load.Input, scpOpts.ImageName, true) // execute the command from line 207 with ToFrom defaulting to true
+			if err != nil {
+				return err
+			}
+			fmt.Println(rep)
+		}
+	} else { // else native load
+		rep, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(rep.Names, ","))
+	}
+	defer os.Remove(f.Name())
+	return nil
+}
+
+// execCommand takes ssh information and an image file path on the client, connects via ssh, executes
+// an scp command and then calls podman image load on the remote machine given the new scp filepath
+func execCommand(iden string, uri *urlP.URL, imageFileIn string, imageFileOut string, image string, ToFrom bool) (string, error) {
```

Comment:
Naming the `ToFrom` parameter using a leading upper-case is unusual; its uses look like global Go variables instead of parameter uses.

Reply:
still outstanding, with the parameter named `LoadSave`.

Reply:
I don't see how load save is ambiguous? I'm open for suggestions to rename it to something else but this name just is meant to show you that it is the boolean that differentiates between loading and saving

Reply:
Changed it to `LoadOrSave` that follows the logic nicely.

Reply:
- What I originally meant that local variables and parameters are, in Go, usually not named with a leading upper-case. If I see `LoadSave`, I go looking for a package-global (package-public, even) variable/function, like `Main`, not in the parameter list. (Sure, an IDE helps with that, once I am already confused.)
- Secondarily, `loadSave` absolutely _is_ ambiguous. What does `true` mean and what does `false` mean? That’s not really fixable by renaming, though, “loadNotSave` is not ambiguous but is still very clumsy, and non-obvious in the caller. It would be possible to do this with an enum (`remoteOperation{Load,Save}`), but much cleaner is to split this into two functions (`saveFromRemote`/`loadToRemote`), as already suggested several times, and eliminate the parameter entirely.

Reply:
I see what you are saying about the variable and will change it to lowercase. My only problem with making two functions is code re usage. I will have to re declare and establish the ssh connection in each function since I think declaring it inside of `scp()` does not make sense. I really think any major code re-working of this kind should wait until we merge this. I am going to continue working on this after it is merged this is merely a v1. I feel like we can rework the functions forever, there are always improvements!